engine/app/components/citizens_advice_components/input.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  class Input < Base
+    attr_reader :name, :label, :error_message, :hint, :width, :value
+
+    def initialize(name:, label:, type:, options: nil)
+      super
+      @name = name
+      @label = label
+      @type = type
```

Comment:
> making type required and not providing a default value of text (want people to think about what inputs they are creating more (eg email, search, number etc) and not just make everything a text field
> not validating type (although there are only c. 20 so I could still do this)

I think in this case what you have here would work. A named argument should throw an error if one is not provided. We could probably write a custom error message though if you want to guide people more.

Reply:
100% agree with the point about making people think about what type they use.

Reply:
Yeah I think I'll enumerate the options here and then have a custom error message referencing mdn or smth 👍🏻